src/System.Private.CoreLib/src/System/Threading/ThreadPool.Unix.cs

```diff
@@ -22,103 +22,48 @@ public bool Unregister(WaitHandle waitObject)
         {
-            // Not supported at present
-            return false;
+            return ClrThreadPool.SetMaxThreads(workerThreads);
         }
 
         public static void GetMaxThreads(out int workerThreads, out int completionPortThreads)
         {
-            // Note that worker threads and completion port threads share the same thread pool.
-            // The total number of threads cannot exceed MaxThreadCount.
-            workerThreads = MaxThreadCount;
-            completionPortThreads = MaxThreadCount;
+            completionPortThreads = workerThreads = ClrThreadPool.GetMaxThreads();
         }
 
         public static bool SetMinThreads(int workerThreads, int completionPortThreads)
         {
-            // Not supported at present
-            return false;
+            return ClrThreadPool.SetMinThreads(workerThreads);
         }
 
         public static void GetMinThreads(out int workerThreads, out int completionPortThreads)
         {
-            // All threads are pre-created at present
-            workerThreads = MaxThreadCount;
-            completionPortThreads = MaxThreadCount;
+            completionPortThreads = workerThreads = ClrThreadPool.GetMinThreads();
         }
 
         public static void GetAvailableThreads(out int workerThreads, out int completionPortThreads)
         {
-            // Make sure we return a non-negative value if thread pool defaults are changed
-            int availableThreads = Math.Max(MaxThreadCount - ThreadPoolGlobals.workQueue.numWorkingThreads, 0);
-
-            workerThreads = availableThreads;
-            completionPortThreads = availableThreads;
+            completionPortThreads = workerThreads = ClrThreadPool.GetAvailableThreads();
         }
 
         /// <summary>
         /// This method is called to request a new thread pool worker to handle pending work.
         /// </summary>
         internal static void QueueDispatch()
         {
-            // For simplicity of the state management, we pre-create all thread pool workers on the first
-            // request and then use the semaphore to release threads as new requests come in.
-            if ((s_workerCount == 0) && Interlocked.Exchange(ref s_workerCount, MaxThreadCount) == 0)
-            {
-                for (int i = 0; i < MaxThreadCount; i++)
-                {
-                    if (!Interop.Sys.RuntimeThread_CreateThread(IntPtr.Zero /*use default stack size*/,
-                        AddrofIntrinsics.AddrOf<Interop.Sys.ThreadProc>(ThreadPoolDispatchCallback), IntPtr.Zero))
-                    {
-                        throw new OutOfMemoryException();
-                    }
-                }
-            }
-
-            // Release one thread to handle the new request
-            s_semaphore.Release(1);
+            ClrThreadPool.WorkerThread.MaybeAddWorkingWorker();
+            // TODO: Ensure gate thread is running here.
```

Comment:
It would be better to call some method on ClrThreadPool, like RequestThread, and have that do this work. The implementation details of ClrThreadPool should not be exposed here.